airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/file/parquet/ParquetWriter.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.file.parquet
+
+import java.io.Closeable
+import java.io.OutputStream
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.avro.AvroParquetWriter
+import org.apache.parquet.hadoop.ParquetWriter as ApacheParquetWriter
+import org.apache.parquet.io.OutputFile
+import org.apache.parquet.io.PositionOutputStream
+
+class ParquetWriter(private val writer: ApacheParquetWriter<GenericRecord>) : Closeable {
+    fun write(record: GenericRecord) = writer.write(record)
+    override fun close() = writer.close()
+}
+
+fun OutputStream.toParquetWriter(avroSchema: Schema): ParquetWriter {
+    // Custom OutputFile implementation wrapping the OutputStream
+    val outputFile =
+        object : OutputFile {
+            var position: Long = 0
+            override fun create(blockSizeHint: Long) =
+                object : PositionOutputStream() {
```

Comment:
we should definitely do some more testing on this, but assuming it works, this is cool!

Reply:
My gut instinct is that it's wrong, it should be tracking the size of the converted data, not the written data. My gut is that it works because in practice it's just increasing padding size.